roles/reportportal/tasks/main.yaml

```
@@ -0,0 +1,79 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if mandatory variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_reportportal_datarouter_url is defined and cifmw_reportportal_datarouter_url != ""
+      - cifmw_reportportal_datarouter_username is defined and cifmw_reportportal_datarouter_username != ""
+      - cifmw_reportportal_datarouter_password is defined and cifmw_reportportal_datarouter_password != ""
+      - cifmw_reportportal_project is defined and cifmw_reportportal_project != ""
+      - cifmw_reportportal_droute_client_url is defined and cifmw_reportportal_droute_client_url != ""
+
+- name: Url for downloading data router client
+  ansible.builtin.set_fact:
+    droute_url: >-
+      {{
+        (cifmw_reportportal_droute_client_url,
+         cifmw_reportportal_droute_version,
+         cifmw_reportportal_droute_binary) |
+        path_join
+      }}
+
+- name: Make sure the droute directory exists
+  ansible.builtin.file:
+    path: "{{ cifmw_reportportal_droute_dir }}"
+    state: directory
+
+- name: Get the Data router client
+  ansible.builtin.get_url:
+    url: "{{ droute_url }}"
+    dest: "{{ cifmw_reportportal_droute_dir }}"
+    backup: true
+    mode: 0755
```

Mode should be declared as string (since it's an octal number)

```suggestion
    mode: "0755"
```

---

Done